compute_endpoint/globus_compute_endpoint/endpoint/config/utils.py

```diff
@@ -281,14 +287,12 @@ def render_config_user_template(
     import jinja2  # Only load package when called by EP manager
     from jinja2.sandbox import SandboxedEnvironment
 
-    _user_opts = user_opts or {}
+    _user_opts = _sanitize_user_json(user_opts or {})
     _validate_user_opts(_user_opts, user_config_schema)
-    _user_opts = _sanitize_user_opts(_user_opts)
```

Comment:
I'm concerned about the reversed order of operations, as sanitize can have a side effect of changing the data.  In fact, if I'm correct that the order of operations does matter, this implies that we have a missing test to verify this.